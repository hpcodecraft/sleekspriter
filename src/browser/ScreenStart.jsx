var ScreenStart = React.createClass({
  render: function() {
    return (
      <section className="screen start">
        <div className="splash">
          <div className="inner">
            <div className="logo">@@name</div>
            <ul>
              <li><a onClick={this.newFile}>New file</a></li>
            </ul>
          </div>
        </div>
        <div className="area statusbar">
          <div className="bar">

          </div>
        </div>
      </section>
    )
  },
  newFile: function() {
    channel.gui.publish('screen.select', {target: 'paint'});
    channel.gui.publish('modal.show', {component: ModalNewFile});
  },
});