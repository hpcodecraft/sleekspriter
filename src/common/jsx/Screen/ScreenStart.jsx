var ScreenStart = React.createClass({
  mixins: [FluxMixin, TouchMixin],
  render: function() {
    return (
      <section className="screen start" onDrop={this.handleDrop}>
        <div className="splash">
          <div className="inner">
            <div className="logo">@@name</div>
            <ul>
              <li><a onClick={this.handleClick.bind(this, this.newFile)} onTouchStart={this.handleTouch.bind(this, this.newFile)}>New file</a></li>
              <li><a onClick={this.handleClick.bind(this, this.openFile)} onTouchStart={this.handleTouch.bind(this, this.openFile)}>Open file</a></li>
            </ul>
          </div>
        </div>
        <div className="area statusbar">
          <div className="bar">
            Tip: Drop .pixels files on this screen to open them directly.
          </div>
        </div>
      </section>
    )
  },
  shouldComponentUpdate: function() {
    switch(stateHistory.last.action) {
      case 'CURSOR_SET':
      case 'TOOL_SELECT':
      case 'SETTINGS_GRID':
      case 'ZOOM_SELECT':
      case 'FRAME_SELECT':
      case 'BRIGHTNESSTOOL_MODE':
      case 'BRIGHTNESSTOOL_INTENSITY':
      case 'COLOR_BRUSH':
      case 'PALETTE_SELECT':
      case 'LAYER_SELECT':
      case 'LAYER_TOP_SELECT':
      case 'BOX_FOLD':
      case 'BOX_PREVIEW_TOGGLE':
      case 'ONION_TOGGLE':
      case 'ONION_MODE':
      case 'ONION_FRAME':
      case 'SELECTION_START':
      case 'SELECTION_RESIZE':
      case 'SELECTION_PREVIEW':
      case 'SELECTION_END':
      case 'SELECTION_CLEAR':
      case 'EXPORT_PART':
      case 'EXPORT_FRAME':
      case 'EXPORT_ANIMATION':
      case 'EXPORT_ZOOM':
      case 'EXPORT_FORMAT':
      case 'EXPORT_STATUS':
        return false;

      default:
        return true;
    }
  },
  newFile: function() {
    this.getFlux().actions.tabSelect('paint');
    this.getFlux().actions.modalShow(ModalNewFile);
  },
  openFile: function() {
    platformUtils.showOpenFileDialog();
  },
  handleDrop: function(e) {
    e.preventDefault();
    if(e.dataTransfer.files.length >= 1) {
      var name = e.dataTransfer.files[0].name,
          suffix = name.substr(name.lastIndexOf('.')+1);

      if(suffix === 'pixels') platformUtils.loadFile(e.dataTransfer.files[0].path);
      else alert('Error: Could not read file format');
    }
  },
});
