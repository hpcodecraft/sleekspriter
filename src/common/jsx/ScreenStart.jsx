var ScreenStart = React.createClass({
  render: function() {
    return (
      <section className="screen start">

      </section>
    )
  },
});