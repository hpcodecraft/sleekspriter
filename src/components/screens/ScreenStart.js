import React from "react";
import { t } from "../../utils";

class ScreenStart extends React.Component {
  render() {
    return (
      <section className="screen start">
        <div className="splash">
          <div className="inner">
            <div className="logo">@@name</div>
            <ul>
              <li><a onClick={::this.newFile}>{t("New file")}</a></li>
              <li><a onClick={::this.openFile}>{t("Open file")}</a></li>
            </ul>
          </div>
        </div>
        <div className="area statusbar">
          <div className="bar">@@version</div>
        </div>
      </section>
    );
  }

  newFile() {
    this.props.modalShow("ModalNewFile");
  }

  openFile() {
    this.props.modalShow("ModalLoadFile");
  }
}

export default ScreenStart;
