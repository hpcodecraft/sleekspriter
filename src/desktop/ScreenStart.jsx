var ScreenStart = React.createClass({
  render: function() {
    return (
      <section className="screen start" onDrop={this.handleDrop}>
        <div className="splash">
          <div className="inner">
            <div className="logo">@@app</div>
            <ul>
              <li><a onClick={this.newFile}>New file</a></li>
              <li><a onClick={this.openFile}>Open file</a></li>
            </ul>
          </div>
        </div>
        <div className="area statusbar">
          <div className="bar">
            Tip: Drop .pixels files on this screen to open them directly.
          </div>
        </div>
      </section>
    )
  },
  newFile: function() {
    channel.publish('screen.select', {target: 'paint'});
    channel.publish('modal.show', {component: ModalNewFile});
  },
  openFile: function() {
    clickInput('fileOpen');
  },
  handleDrop: function(e) {
    e.preventDefault();
    if(e.dataTransfer.files.length >= 1) {
      var name = e.dataTransfer.files[0].name,
          suffix = name.substr(name.lastIndexOf('.')+1);
      if(suffix === 'pixels') file.load(e.dataTransfer.files[0].path, fileLoaded);
      else alert('Error: Could not read file format');
    }
  },
});