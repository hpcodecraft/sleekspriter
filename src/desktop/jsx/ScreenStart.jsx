var ScreenStart = React.createClass({
  mixins: [FluxMixin],
  render: function() {
    return (
      <section className="screen start" onDrop={this.handleDrop}>
        <div className="splash">
          <div className="inner">
            <div className="logo">@@name</div>
            <ul>
              <li><a onClick={this.newFile}>New file</a></li>
              <li><a onClick={this.openFile}>Open file</a></li>
            </ul>
          </div>
        </div>
        <div className="area statusbar">
          <div className="bar">
            Tip: Drop .pixels files on this screen to open them directly.
          </div>
        </div>
      </section>
    )
  },
  newFile: function() {
    this.getFlux().actions.tabSelect('paint');
    this.getFlux().actions.modalShow(ModalNewFile);
  },
  openFile: function() {
    platformUtils.showOpenFileDialog();
  },
  handleDrop: function(e) {
    e.preventDefault();
    if(e.dataTransfer.files.length >= 1) {
      var name = e.dataTransfer.files[0].name,
          suffix = name.substr(name.lastIndexOf('.')+1);

      if(suffix === 'pixels') platformUtils.loadFile(e.dataTransfer.files[0].path);
      else alert('Error: Could not read file format');
    }
  },
});