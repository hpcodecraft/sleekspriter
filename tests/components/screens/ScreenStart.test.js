import React from "react";
import ScreenStart from "components/screens/ScreenStart";

describe("ScreenStart", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      modalShow: jest.fn()
    };

    wrapper = shallow(<ScreenStart {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("newFile", () => {
    it("shows new file modal window", () => {
      wrapper
        .find("a")
        .at(0)
        .simulate("click");
      expect(props.modalShow).toBeCalledWith("ModalNewFile");
    });
  });

  describe("openFile", () => {
    it("shows load file modal window", () => {
      wrapper
        .find("a")
        .at(1)
        .simulate("click");
      expect(props.modalShow).toBeCalledWith("ModalLoadFile");
    });
  });
});
