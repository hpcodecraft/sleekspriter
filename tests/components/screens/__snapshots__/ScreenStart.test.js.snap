// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenStart should render correctly 1`] = `
<section
  class="screen start"
>
  <div
    class="splash"
  >
    <div
      class="inner"
    >
      <div
        class="logo"
      >
        SleekSpriter
      </div>
      <ul>
        <li>
          <a>
            New file
          </a>
        </li>
        <li>
          <a>
            Open file
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="area statusbar"
  >
    <div
      class="bar"
    >
      0.2.16
    </div>
  </div>
</section>
`;
